Update getting started docs (#1214)
nuts-foundation/nuts-node: docs/pages/getting-started/3-configure-your-node.rst

To generate the certificate for your own node you need the ``https://github.com/nuts-foundation/nuts-development-network-ca`` repository. It contains handy scripts and the needed key material.

Your node only accepts requests from other nodes which use a certificate issued by one of the trusted CAs. Which CAs to trust is configured by providing a truststore file. The truststore is a PEM file which contains a set of one or more certificates from CAs which the network participants all decided on to trust.
To learn more about how a Nuts network uses certificates, see the specification `RFC008 <https://nuts-foundation.gitbook.io/drafts/rfc/rfc008-certificate-structure>`_.

To generate certificates for the `development` network perform the following steps:

.. code-block:: shell

  git clone https://github.com/nuts-foundation/nuts-development-network-ca
  cd nuts-development-network-ca
  ./issue-cert.sh development nuts.yourdomain.example

> This results in 3 files:

Reviewer: has the script been changed? The CA had to be copied from the ca directory manually.

Reviewer: what about pointing to the README of that repo? Now both have to be kept in sync.

Author: The script did indeed changed. Now it copies the $NETWORK/ca to ./truststore-$NETWORK.pem. So it is clear what the file contains and for which network.

* ``nuts.yourdomain.example-development.key`` The private key for the node.
* ``nuts.yourdomain.example-development.pem`` The certificate for the node.
* ``truststore-development.pem`` The truststore for this (development) network.

Bootstrap nodes
===============

> A bootstrap node is just a normal Nuts node which is available for other nodes to connect to. When you want to join a network, you must approach another network participant and ask for its public endpoint. After connecting, this node will sent an address list of other connected nodes (this requires a Network Identity and NutsComm endpoint, see further down for more information). Your node must connect to the GRPC endpoint which is usually configured on the ``5555`` port.

Reviewer: A bit of ambiguity: do you need a NutsComm endpoint to receive the addresses or do they need one?

> Consult the community on `Slack <https://nuts-foundation.slack.com/>`_ to find out which public bootstrap nodes are available to connect to your network of choice.

Reviewer: Add channel.

The following is needed to connect a Nuts node to a network:

1. A runnable node.
2. A network you want to join.
3. A TLS client- and server certificate which is accepted by the other nodes in the network (e.g. PKIoverheid).
4. The public address of one or more remote nodes you'd like to use as bootstrap nodes.

> 5. A node Identity to identify yourself in te network (optionally)

Reviewer 2: ```suggestion
5. A node identity (node DID) to identify yourself in the network, so you can send/receive private transactions.
```

Networks
========

> A network is a set of nodes who trust each other. They share a single root transaction. Trust is put in one or more network authorities who issue certificates (Certificate Authorities or CAs). Each node can be configured with a certificate and a set of trusted Certificate Authorities.

Author: It's trust chain right? I trust a root who endorses a node so I trust the node.

> A network is a set of nodes who trust each other. They share a single root transaction. Trust is put in one or more network authorities who issue certificates (Certificate Authorities or CAs). Each node can be configured with a certificate and a set of trusted Certificate Authorities.

Reviewer 2: Then, what kind of trust do we talk about here? What is derived from the trust?

What "trust" comes from a PKI CA is commonly known (since it's a established concept), here it is not. What does the sentence add? I'd say it only adds confusing because we also have vendors trust each other issuing specific VC types.

Your node only accepts requests from other nodes which use a certificate issued by one of the trusted CAs. Which CAs to trust is configured by providing a truststore file. The truststore is a PEM file which contains a set of one or more certificates from CAs which the network participants all decided on to trust.
To learn more about how a Nuts network uses certificates, see the specification `RFC008 <https://nuts-foundation.gitbook.io/drafts/rfc/rfc008-certificate-structure>`_.

To generate certificates for the `development` network perform the following steps:

.. code-block:: shell

  git clone https://github.com/nuts-foundation/nuts-development-network-ca
  cd nuts-development-network-ca
  ./issue-cert.sh development nuts.yourdomain.example

This results in 3 files:

* ``nuts.yourdomain.example-development.key`` The private key for the node.
* ``nuts.yourdomain.example-development.pem`` The certificate for the node.
* ``truststore-development.pem`` The truststore for this (development) network.

Bootstrap nodes
===============

> A bootstrap node is just a normal Nuts node which is available for other nodes to connect to. When you want to join a network, you must approach another network participant and ask for its public endpoint. After connecting, you receive a copy of the current state of the network. These transactions contain endpoints of other nodes. After a reboot, your node will try to connect to other nodes discoverd in the network. Your node must connect to the others GRPC endpoint which is usually configured on the ``5555`` port.

Reviewer 2: ```suggestion
A bootstrap node is just a normal Nuts node which is available for other nodes to connect to. When you want to join a network, you must approach another network participant and ask for its public (gRPC) endpoint. After connecting, you receive a copy of the current state of the network. These transactions contain endpoints of other nodes. After a reboot, your node will try to connect to other nodes discoverd in the network. Your node will have to connect to the bootstrap node's gRPC endpoint which is configured on port ``5555`` by default.
```

Bootstrap nodes
===============

> A bootstrap node is just a normal Nuts node which is available for other nodes to connect to. When you want to join a network, you must approach another network participant and ask for its public endpoint. After connecting, this node will sent an address list of other connected nodes (this requires you to configure a Network Identity and NutsComm endpoint, see further down for more information). Your node must connect to the others GRPC endpoint which is usually configured on the ``5555`` port.

Reviewer: Ok, ambiguity is gone but now it's incorrect. You don't need an identity to receive DID Documents and/or connect to other nodes.

where the domain part (e.g. ``nuts.nl``) matches (one of) the DNS SANs in your node's TLS certificate.
See "Node Discovery" below for more information on registering the ``NutsComm`` endpoint.

.. note::

>     After registering of the ``nodedid``, you need to reboot your node in order to receive private transactions.

Reviewer 2: ```suggestion
    After registering ``nodedid`` you need to reboot your node in order have your connections authenticated, which is required to receive private transactions.
```
